packages/block-editor/src/components/inserter/menu.js

```diff
@@ -46,18 +47,21 @@ function InserterMenu( {
 		isAppender,
 		selectBlockOnInsert: __experimentalSelectBlockOnInsert,
 	} );
-	const { hasPatterns } = useSelect(
-		( select ) => {
-			const { getSettings } = select( 'core/block-editor' );
-			return {
-				hasPatterns: !! getSettings().__experimentalBlockPatterns
-					?.length,
-			};
-		},
-		[ isAppender, clientId, rootClientId ]
-	);
+	const { hasPatterns, hasReusableBlocks } = useSelect( ( select ) => {
+		const {
+			__experimentalBlockPatterns,
+			__experimentalReusableBlocks,
+		} = select( 'core/block-editor' ).getSettings();
+
+		return {
+			hasPatterns: !! __experimentalBlockPatterns?.length,
+			hasReusableBlocks: !! __experimentalReusableBlocks?.length,
```

Comment:
Should I instead use the following?

```js
select( 'core' ).getEntityRecords( 'postType', 'wp_block' )?.length
```

I'm not sure which makes more sense or which is faster.

Reply:
no, you used the right one here. The block-editor package should use the REST API direclty.